### PR TITLE
MIC-86 referral area-of-work screen

### PR DESCRIPTION
**Pull request by bill-moj**

### CODE REVIEW

File: server/journeys/referral/area-of-work/controller.ts [hunk 1]
```diff
@@ -0,0 +1,31 @@
...
+  GET = async (req: Request, res: Response): Promise<void> => {
+    req.journeyData.referral ??= {}
+
+    const areaOfWorkOptions = [
+      {
+        value: '',
+        text: 'Select area',
+        selected: !req.journeyData.referral.refererArea,
+      },
+      ...(await this.csipApiService.getReferenceData(req, 'area-of-work')).map(refData => ({
```
Comment (iterative-it) [inline server/journeys/referral/area-of-work/controller.ts:16]: Would be nice to move all of this down into the service layer so you end up with only `await this.csipApiService.getReferenceData(req, 'area-of-work', 'Select area'))` call with all the mapping and the insertion of the default option done in the service. That way that can be unit tested in isolation

Comment (bill-moj) [reply server/journeys/referral/area-of-work/controller.ts:16]: we still have use for ReferenceData[], while this function is generating govukSelect's option items which is slightly different.
i'd prefer to make it a base controller class function to convert ReferenceData[] to govukSelect options.

Comment (iterative-it) [reply server/journeys/referral/area-of-work/controller.ts:16]: As long as this code is shared somewhere, it'd fine. Base controller class function makes sense to me

File: server/services/csipApi/csipApiClient.ts [hunk 2]
```diff
@@ -0,0 +1,17 @@
+import RestClient from '../../data/restClient'
+import config from '../../config'
+import { ReferenceData, ReferenceDataType } from '../../@types/csip/csipApiTypes'
+
+export default class CsipApiClient {
+  private readonly restClient: RestClient
+
+  constructor(token: string) {
+    this.restClient = new RestClient('Prison API', config.apis.csipApi, token)
```
Comment (iterative-it) [inline server/services/csipApi/csipApiClient.ts:9]: Tiny copy paste error, CSIP API rather than Prison API

Comment (bill-moj) [reply server/services/csipApi/csipApiClient.ts]: fixed

File: server/journeys/referral/area-of-work/schemas.ts [hunk 3]
```diff
@@ -0,0 +1,20 @@
...
+  async (req: Request): Promise<z.ZodTypeAny> => {
+    const areaRefDataMap = new Map(
+      (await csipApiService.getReferenceData(req, 'area-of-work')).map(itm => [itm.code, itm]),
+    )
+
+    return z.object({
+      areaOfWork: z
+        .string({ message: 'Select your area of work' })
+        .min(1, 'Select your area of work')
+        .max(12, 'Select your area of work')
```
Comment (iterative-it) [inline server/journeys/referral/area-of-work/schemas.ts:16]: I don't think the min max restrictions are needed any more as the refine bit below should cover it?

File: server/config.ts [hunk 4]
```diff
@@ -89,13 +89,21 @@ export default {
...
+        response: Number(get('CSIP_API_TIMEOUT_RESPONSE', 20000)),
+        deadline: Number(get('CSIP_API_TIMEOUT_DEADLINE', 20000)),
+      },
+      agent: new AgentConfig(Number(get('CSIP_API_TIMEOUT_RESPONSE', 20000))),
+    },
     prisonerSearchApi: {
       url: get('PRISONER_SEARCH_API_URL', 'http://localhost:8082', requiredInProduction),
       timeout: {
         response: Number(get('PRISONER_SEARCH_API_TIMEOUT_SECONDS', 20000)),
-        deadline: Number(get('PRISONER_SEARCH_API_TIMEOUT_SECONDS', 20000)),
```
Comment (iterative-it) [inline server/config.ts:96]: (Minor) all the others use `_RESPONSE` not `_SECONDS` so `PRISONER_SEARCH_API_TIMEOUT_RESPONSE` especially as the value isn't seconds

File: server/journeys/referral/area-of-work/schemas.ts [hunk 5]
```diff
@@ -0,0 +1,20 @@
...
+        .refine(val => areaRefDataMap.has(val), 'Select your area of work')
```
[the remark this answers is not in the document]
Comment (bill-moj) [reply server/journeys/referral/area-of-work/schemas.ts:17]: make sense. likely for the other screens to take a similar approach of using one generic error message for all kind of validation errors?

File: server/journeys/referral/area-of-work/tests.test.ts [hunk 6]
```diff
@@ -0,0 +1,154 @@
...
+
+    const result = await request(appWithError)
+      .get(`/${uuid}/referral/area-of-work`)
+      .expect(200)
+      .expect('Content-Type', /html/)
+
+    const html = createTestHtmlElement(result.text)
+    expect(getByText(html, 'Which area do you work in?')).toBeVisible()
+    expect(getByText(html, 'There is a problem')).toBeVisible()
+    expect(getByText(html, 'Select your area of work')).toBeVisible()
```
Comment (iterative-it) [inline server/journeys/referral/area-of-work/tests.test.ts:77]: I liked James' `const error = getByRole(topLevelElement, 'link', { name: onBehalfOfError }) expect(error).toBeVisible()` test for this as it asserts that the error summary links to the correct form element

**Review by iterative-it — APPROVED** [review]
LGTM

**Pull request merged**